tck/src/test/java/org/reactivestreams/tck/IdentityProcessorVerificationTest.java

```diff
@@ -115,6 +126,153 @@ public void required_spec104_mustCallOnErrorOnAllItsSubscribersIfItEncountersANo
+              try {
+                subscriber.onComplete();
+              } catch (Exception ex) {
+                env.flop(ex, String.format("Calling onComplete on [%s] should not throw! See https://github.com/reactive-streams/reactive-streams-jvm#2.13", subscriber));
+              }
+            }
+          }
+
+          private Subscription createSubscription(CountDownLatch awaitLatch, final Subscriber<? super Integer> s, final AtomicLong demand) {
+            final MySubscription sub = new MySubscription(awaitLatch, s, demand);
+            subs.add(sub);
+            subscribers.add(s);
+            return sub;
+          }
+
+          final class MySubscription implements Subscription {
+            private final CountDownLatch awaitLatch;
+            private final Subscriber<? super Integer> s;
+            private final AtomicLong demand;
+
+            public MySubscription(CountDownLatch awaitTwoLatch, Subscriber<? super Integer> s, AtomicLong demand) {
+              this.awaitLatch = awaitTwoLatch;
+              this.s = s;
+              this.demand = demand;
+            }
+
+            @Override
+            public void request(final long n) {
+              new Thread(new Runnable() {
+                @Override
+                public void run() {
+                  demand.addAndGet(n); // naive, but good enough for the test
+                  awaitLatch.countDown();
+                  try {
+                    awaitLatch.await(env.defaultTimeoutMillis(), TimeUnit.MILLISECONDS);
+                    while (demand.getAndDecrement() > 0) {
+                      upstreamSubscription.request(1);
+                    }
+                  } catch (InterruptedException e) {
+                    env.flop(e, "Interrupted while awaiting for all downstreams to signal some demand.");
+                  }
+
+                }
+              }).start();
+            }
+
+            @Override
+            public void cancel() {
+              demand.set(Long.MIN_VALUE); // naive but OK for this test
+            }
+
+            @Override
+            public String toString() {
+              return String.format("IdentityProcessorVerificationTest:MySubscription(%s, demand = %s)", s, demand);
+            }
+          }
+        };
+      }
+
+      @Override
+      public ExecutorService publisherExecutorService() {
+        return ex;
+      }
+
+      @Override
+      public Integer createElement(int element) {
+        return element;
+      }
+
+      @Override
+      public Publisher<Integer> createHelperPublisher(long elements) {
+        return new Publisher<Integer>() {
+          @Override
+          public void subscribe(final Subscriber<? super Integer> s) {
+            s.onSubscribe(new NoopSubscription() {
+              @Override
+              public void request(long n) {
+                for (int i = 0; i < 10; i++) {
+                  s.onNext(i);
```

Comment:
This is not legal afaict since it allows unbounded recursion between request and onNext?